### PR TITLE
Vendor OpenTelemetry.NET Instrumentation Libraries into Azure.Monitor.OpenTelemetry.AspNetCore

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
### Clone the OpenTelemetry.NET Repository

```bash
git clone https://github.com/open-telemetry/opentelemetry-dotnet.git
```

### Fetch the Tags

```bash
git fetch --tags
```

### Check Out the Tag for the Recent Release

```bash
git checkout tags/1.5.1-beta.1
```

### Create a New Branch

While you are at this tag, create a new branch:

```bash
git checkout -b my_local_branch
```

### Copy Folders

Copy the following folders from `OpenTelemetry.NET` to the `Vendoring` folder of `Azure.Monitor.OpenTelemetry.AspNetCore`:

- `OpenTelemetry.Instrumentation.AspNetCore`
- `OpenTelemetry.Instrumentation.Http`
- `OpenTelemetry.Instrumentation.SqlClient`
- `Shared`

### Remove Unnecessary Files

Remove the following files:

- \*.csproj
- README.md
- CHANGELOG.MD
- AssemblyInfo.cs

### Additional Changes

- Added #nullable disable to `Azure.Monitor.OpenTelemetry.AspNetCore` (same as the instrumentation library project). Once the nullables are fixed in the instrumentation library, this can be removed.
- Removed existing instrumentation package references.
- Added `OpenTelemetry.Instrumentation.AspNetCore` dependent package reference to `Azure.Monitor.OpenTelemetry.AspNetCore`.
- Modified all instrumentation public APIs to internal.
- Added Azure Copyrights to the top of each file, along with the OpenTelemetry license.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/StatusHelper.cs
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/StatusHelper.cs:1]: This file can be removed.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/SpanHelper.cs
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/SpanHelper.cs:1]: This one too

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentation.cs
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentation.cs:2]: Who is handling the legal review? Who should bless the relicensing from Apache 2.0 to MIT?

**Review by reyang — DISMISSED** [review]
Blocking relicensing until there is evidence of proper legal review.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentation.cs
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentation.cs:2]: I'm following up with CELA on this.

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentation.cs:2]: The response from CELA:

> For the copied files just leave the 3rd party Apache license header as-is and do not add the Microsoft copyright statement.

**Review by vishweshbankwar — APPROVED** [review]
LGTM

**Review by TimothyMothra — APPROVED** [review]
I compared this branch to the source otel branch and confirmed that the files match.
LGTM.

**Comment by reyang** [comment]
My last question (not blocking this PR) - if the user chooses to use the instrumentation from OpenTelemetry (e.g. adding ASP.NET Core instrumentation) on top of this solution, what would happen?

**Comment by rajkumar-rangaraj** [comment]
> My last question (not blocking this PR) - if the user chooses to use the instrumentation from OpenTelemetry (e.g. adding ASP.NET Core instrumentation) on top of this solution, what would happen?

With our current vendoring approach, we have changed all the public APIs of the instrumentation libraries to internal. This means that customers can't modify the instrumentation options using just the distro. To alter the instrumentation options for ASP.NET Core, HTTP Client, and Sql Client instrumentation, customers need to reference the OpenTelemetry instrumentation package and use an API like AddAspNetCoreInstrumentation to adjust its options. In subsequent PRs, we plan to disable internal instrumentation if a public one is detected. This is an interim solution until we take reference to the stable version of instrumentation packages.

**Comment by reyang** [comment]
> In subsequent PRs, we plan to disable internal instrumentation if a public one is detected. This is an interim solution until we take reference to the stable version of instrumentation packages.

Got you, awesome! 👍

**Review by reyang — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/OpenTelemetry.Instrumentation.Http/HttpClient_TracerProviderBuilderExtensions.cs
Comment (lmolkova) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/OpenTelemetry.Instrumentation.Http/HttpClient_TracerProviderBuilderExtensions.cs:65]: do we need to check if user has brought up beta dependency and configured custom HTTP instrumentation options and back off ?

(same question for other instrumentations)

It's totally fine if you want to do it in a separate PR.

**Comment by lmolkova** [comment]
We also need to add a notice to the corresponding file to comply with Apache license

https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/LICENSE#L106-L121

You can find an example here https://github.com/Azure/azure-sdk-for-java/blob/main/NOTICE.txt#L112-L125

**Review by lmolkova — APPROVED** [review]
LGTM, but please update NOTICE file in the root of the repo before merging.

**Review by lmolkova — APPROVED** [review]

**Pull request merged**
